src/components/Pages/index.js

```diff
@@ -192,37 +200,43 @@ class VmCreatePage extends React.Component {
     props.getAvailableCDImages()
   }
 
-  componentDidUpdate () {
-    if (!this.props.isFilteredClusters && this.props.clusterSize) {
+  render () {
+    if (!this.props.isAnyUsableCluster && this.props.clusterSize) {
```

Comment:
Is there any reason to move it from `componentDidUpdate`? Bacause `render` method must be pure, and dispatching actions this is not very pure solution. 
https://reactjs.org/docs/react-component.html#render

Reply:
I did it specifically this way since `componentDidUpdate` does not run during the first render cycle and the page was flashing before kicking the user back to the main page.  Humm...I'm not 100% sure which way to go to keep `render` pure.

What do you think about having `render` do the test and just returning null and then putting the dispatch back in `componentDidUpdate`?

Reply:
I'm updating following that direction...

Reply:
> What do you think about having render do the test and just returning null and then putting the dispatch back in componentDidUpdate?

Yes, it sounds fine.